File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableBoolean.java
package app.android.ttp.mikazuki.yoshinani.binding;

import android.databinding.BaseObservable;
import android.support.annotation.Nullable;

import org.parceler.Parcel;

import java.util.Objects;

/**
 * Created by haijimakazuki on 16/01/26.
 */
@Parcel
public class BindableBoolean extends BaseObservable {
    boolean mValue;

    public boolean get() {
        return mValue;
    }

    public void set(@Nullable final boolean value) {
        if (!Objects.equals(mValue, value)) {
            mValue = value;
            notifyChange();
        }
    }
}

File: app/src/main/java/app/android/ttp/mikazuki/yoshinani/binding/BindableBoolean.kt
package app.android.ttp.mikazuki.yoshinani.binding

import android.databinding.BaseObservable

import org.parceler.Parcel

import java.util.Objects

/**
 * Created by haijimakazuki on 16/01/26.
 */
@Parcel
class BindableBoolean : BaseObservable() {
    internal var mValue: Boolean = false

    fun get(): Boolean {
        return mValue
    }

    fun set(value: Boolean) {
        if (mValue != value) {
            mValue = value
            notifyChange()
        }
    }
}